examples/seismic/abc_methods/01_introduction.ipynb

```diff
@@ -0,0 +1,1192 @@
+    "compx  =  x1-x0\n",
+    "z0     =  0.\n",
+    "z1     =  1000.\n",
+    "compz  =  z1-z0;\n",
+    "hx     =  (x1-x0)/(nptx-1)\n",
+    "hz     =  (z1-z0)/(nptz-1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once the spatial parameters are defined, we start the construction of the *spatial grid* using the Devito DSL. The required parameters are:\n",
+    "\n",
+    "- *origin* is the point of origin of the grid;\n",
+    "- *extent* is the vector with the lengths of the domain in each of the directions;\n",
+    "- *shape* is the vector with the number of points in each direction;\n",
+    "- *spacing* is the vector with the spacing in each of the directions;\n",
+    "\n",
+    "These parameters are set as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "origin  = (x0,z0)\n",
+    "extent  = (compx,compz)\n",
+    "shape   = (nptx,nptz)\n",
+    "spacing = (hx,hz)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once the mesh parameters are built, we can define the structures that represent $\\Omega$ subdomains, that is, particular regions of $\\Omega $, named *subdomains*. In the present notebook, there is no need to split  the domain in particular subregions, so we define a single *subdomain* that correponds to the full domain $\\Omega$. This *subdomain* is built with the following command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class d0domain(SubDomain):\n",
+    "    name = 'd0'\n",
+    "    def define(self, dimensions):\n",
+    "        x, z = dimensions\n",
+    "        return {x: z, z: z}\n",
+    "d0_domain = d0domain()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After defining the spatial parameters and *subdomains*, we generate the *spatial grid*:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "grid = Grid(origin=origin, extent=extent, shape=shape, subdomains=(d0_domain))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The velocity field is set bellow, representing four different layers:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "v0 = np.zeros((nptx,nptz))                     \n",
+    "p0 = 0    \n",
+    "p1 = int((1/2)*nptz)\n",
+    "p2 = nptz  \n",
+    "v0[0:nptx,p0:p1] = 1.5\n",
+    "v0[0:nptx,p1:p2] = 2.5"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Below we include a routine to plot the velocity field."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def graph2dvel(vel):\n",
```

Comment:
Could this and other such routines be moved to a python file and imported for use in the notebook?

Reply:
It is completely possible to take these routines to a separate file. We have already been asked to use Devito's plot routines (velocity, displacement, receivers and source). The function to plot the source we use src.show (), but for the other plots we think it would be better to use the routines of our Notebooks as they can be easily adapted to the user's routines.

Reply:
@rhodrin Here is it ok?!